### PR TITLE
Add macro to compute harmonic averages 

### PR DESCRIPTION
**Pull request by albert-de-montserrat**
Adds the set of macros `@harm` to compute harmonic averages in 1/2/3D grids. This family of macros is analogous to the `@av` one.

### CODE REVIEW

File: test/test_FiniteDifferences2D.jl [hunk 1]
```diff
@@ -46,10 +46,16 @@ end
                 @parallel inn!(R, Axxyy) = (@all(R) = @inn(Axxyy); return)
                 @parallel inn_x!(R, Axx) = (@all(R) = @inn_x(Axx); return)
                 @parallel inn_y!(R, Ayy) = (@all(R) = @inn_y(Ayy); return)
+                @parallel idx_x!(R, A)   = (@all(R) = @idx_x(A); return)
```
Comment (omlins) [inline test/test_FiniteDifferences2D.jl:49]: This line with `idx_` as well as all the following does not belong in this PR, does it?

**Review by omlins — CHANGES_REQUESTED** [review]
Thanks for the PR! It looks good to me apart from the two little things I noted.

File: src/FiniteDifferences.jl [hunk 2]
```diff
@@ -49,6 +54,7 @@ macro     d2(A::Symbol)  esc(:( ($A[$ixi+1] - $A[$ixi])  -  ($A[$ixi] - $A[$ixi-
 macro    all(A::Symbol)  esc(:( $A[$ix  ] )) end
 macro    inn(A::Symbol)  esc(:( $A[$ixi ] )) end
 macro     av(A::Symbol)  esc(:(($A[$ix] + $A[$ix+1] )*0.5 )) end
+macro   harm(A::Symbol)  esc(:(2/(1/$A[$ix] + 1/$A[$ix+1]) )) end
```
[the remark this answers is not in the document]
Comment (albert-de-montserrat) [reply src/FiniteDifferences.jl]: Agree, I see slightly better performance with your version, except when `A` is an integer.  I guess that it will be used with floats most of the time anyway.

**Review by omlins — APPROVED** [review]
